src/js/view/controls/components/menu/tizen-menu.js

```diff
@@ -0,0 +1,231 @@
+} from 'utils/dom';
+import { TizenMenuTemplate, TizenSubmenuTemplate } from 'view/controls/templates/menu/menu';
+
+export class TizenMenu extends Menu {
+    constructor(api, model, controlbar, localization) {
+        super('settings', localization.settings, null, localization, TizenMenuTemplate);
+        this.onInteraction = this.onInteraction.bind(this);
+        this.api = api;
+        this.model = model;
+        this.localization = localization;
+        this.visible = false;
+        this.ui = new UI(this.el);
+        this.onTransition = onTransition.bind(this);
+        this.controlbar = controlbar;
+        this.addEventListeners();
+    }
+
+    addEventListeners() {
+        const { model } = this;
+
+        this.on('visibility', this.setVisibility, this);
+        this.on('menuAppended', this.updateControlbarButtons());
+
+        model.on('change:playlistItem', this.onPlaylistItem, this);
+        model.change('audioTracks', this.onAudioTracks, this);
+        model.on('change:currentAudioTrack', this.onAudioTrackIndex, this);
+        model.change('captionsList', this.onCaptionsList, this);
+        model.change('captionsIndex', this.onCaptionsIndex, this);
+
+        this.el.addEventListener('animationend', this.onTransition);
+    }
+
+    setupMenu(menuName, menuTitle, items, onItemSelect, defaultItemIndex, itemOptions) {
+        if (!items || items.length <= 1) {
+            this.removeMenu(menuName);
+            return;
+        }
+        let menu = this.children[menuName];
+    
+    
+        if (!menu) {
+            // We pass null as parent menu to avoid using the standard menu HTML heirarchy.
+            menu = new Menu(menuName, menuTitle, null, this.localization, TizenSubmenuTemplate);
+            menu.parentMenu = this;
+            menu.itemsContainer = new UI(menu.el.querySelector('.jw-settings-submenu-items'));
+            const noop = () => {};
+            // replace open, close and toggle with noops. Submenus always visible in Tizen.
+            menu.open = menu.close = menu.toggle = noop;
+            menu.el.classList.add('jw-settings-submenu-active');
+            // add tizen control listener
+            this.appendMenu(menu);
+        }
+
+        menu.setMenuItems(
+            menu.createItems(items, onItemSelect, itemOptions), 
+            defaultItemIndex
+        );
+    }
+    
+    onInteraction(evt) {
+        const onNavigateMenu = (isUp) => {
+            const next = isUp ? previousSibling(evt.target) : nextSibling(evt.target);
+            if (next) {
+                next.focus({ preventScroll: true });
+            } else {
+                const nextMenu = isUp ? this.children.audioTracks : this.children.captions;
+                const isMenuNext = nextMenu.el.contains(evt.target);
+                if (isMenuNext && nextMenu) {
+                    const items = nextMenu.items;
+                    items[items.length - 1].el.focus({ preventScroll: true });
```

Comment:
If we were scrolling down from the audioTracksMenu to the captionsMenu, would this highlight the last item in the captionsMenu as opposed to the first item?